src/scenario.jl

```diff
@@ -879,12 +879,14 @@ function run_model(domain::Domain, param_set::YAXArray)::NamedTuple
 
         # Update record
         C_cover[tstep, :, :] .= C_t
-        temp_change = abs.(
-            permutedims(
-                reshape(C_cover[tstep, :, :], (n_sizes, n_groups, n_locs)),
-                [2, 1, 3]
-            ) .- cover_copy
+        cover_copy[cover_copy .== 0] .= 1.0
+        temp_change = _group_cover_locs(
+            domain.coral_growth, 
+            C_cover[tstep, :, :]
         ) ./ cover_copy
+
+        # The smallest size class is reconstructed every timestep from the cover
+        temp_change[:, 1, :] .= 1.0
```

Comment:
Is this what you meant?

```suggestion
        # The smallest size class is replaced every timestep by recruitment
        temp_change[:, 1, :] .= 1.0
```